### PR TITLE
Bring the Haskell Kore code in sync with the docs.

### PR DESCRIPTION
**Pull request by virgil-serbanuta**
* \mem -> \in rename
* Changed \in to accept generic patterns on the first position instead of only variables.

### CODE REVIEW

File: src/main/haskell/language-kore/src/Data/Kore/Parser/ParserImpl.hs [hunk 1]
```diff
@@ -629,7 +629,7 @@ BNF definitions:
     | ‘\ceil’ ‘{’ ⟨meta-sort⟩ ‘,’ ⟨meta-sort⟩ ‘}’ ‘(’ ⟨pattern⟩ ‘)’
     | ‘\floor’ ‘{’ ⟨meta-sort⟩ ‘,’ ⟨meta-sort⟩ ‘}’ ‘(’ ⟨pattern⟩ ‘)’
     | ‘\equals’ ‘{’ ⟨meta-sort⟩ ‘,’ ⟨meta-sort⟩ ‘}’ ‘(’ ⟨pattern⟩ ‘,’ ⟨pattern⟩ ‘)’
-    | ‘\mem’ ‘{’ ⟨meta-sort⟩ ‘,’ ⟨meta-sort⟩ ‘}’ ‘(’ ⟨variable⟩ ‘,’ ⟨pattern⟩ ‘)’
+    | ‘\in’ ‘{’ ⟨meta-sort⟩ ‘,’ ⟨meta-sort⟩ ‘}’ ‘(’ ⟨variable⟩ ‘,’ ⟨pattern⟩ ‘)’
```
Comment (traiansf) [inline src/main/haskell/language-kore/src/Data/Kore/Parser/ParserImpl.hs]: `<pattern>`

File: src/main/haskell/language-kore/src/Data/Kore/Parser/ParserImpl.hs [hunk 2]
```diff
@@ -710,7 +710,7 @@ BNF definitions:
     | ‘\ceil’ ‘{’ ⟨object-sort⟩ ‘,’ ⟨object-sort⟩ ‘}’ ‘(’ ⟨pattern⟩ ‘)’
     | ‘\floor’ ‘{’ ⟨object-sort⟩ ‘,’ ⟨object-sort⟩ ‘}’ ‘(’ ⟨pattern⟩ ‘)’
     | ‘\equals’ ‘{’ ⟨object-sort⟩ ‘,’ ⟨object-sort⟩ ‘}’ ‘(’ ⟨pattern⟩ ‘,’ ⟨pattern⟩ ‘)’
-    | ‘\mem’ ‘{’ ⟨object-sort⟩ ‘,’ ⟨object-sort⟩ ‘}’ ‘(’ ⟨variable⟩ ‘,’ ⟨pattern⟩ ‘)’
+    | ‘\in’ ‘{’ ⟨object-sort⟩ ‘,’ ⟨object-sort⟩ ‘}’ ‘(’ ⟨variable⟩ ‘,’ ⟨pattern⟩ ‘)’
```
Comment (traiansf) [inline src/main/haskell/language-kore/src/Data/Kore/Parser/ParserImpl.hs]: Isn't it `<pattern>` here?

File: src/main/haskell/language-kore/src/Data/Kore/Parser/ParserImpl.hs [hunk 3]
```diff
@@ -614,7 +614,7 @@ BNF definitions:
...
```
Comment (traiansf) [inline src/main/haskell/language-kore/src/Data/Kore/Parser/ParserImpl.hs]: `pattern`

File: src/main/haskell/language-kore/src/Data/Kore/Parser/ParserImpl.hs [hunk 4]
```diff
@@ -728,7 +728,7 @@ BNF definitions:
     | ‘\ceil’ ‘{’ ⟨meta-sort⟩ ‘,’ ⟨meta-sort⟩ ‘}’ ‘(’ ⟨pattern⟩ ‘)’
     | ‘\floor’ ‘{’ ⟨meta-sort⟩ ‘,’ ⟨meta-sort⟩ ‘}’ ‘(’ ⟨pattern⟩ ‘)’
     | ‘\equals’ ‘{’ ⟨meta-sort⟩ ‘,’ ⟨meta-sort⟩ ‘}’ ‘(’ ⟨pattern⟩ ‘,’ ⟨pattern⟩ ‘)’
-    | ‘\mem’ ‘{’ ⟨meta-sort⟩ ‘,’ ⟨meta-sort⟩ ‘}’ ‘(’ ⟨variable⟩ ‘,’ ⟨pattern⟩ ‘)’
+    | ‘\in’ ‘{’ ⟨meta-sort⟩ ‘,’ ⟨meta-sort⟩ ‘}’ ‘(’ ⟨variable⟩ ‘,’ ⟨pattern⟩ ‘)’
```
Comment (traiansf) [inline src/main/haskell/language-kore/src/Data/Kore/Parser/ParserImpl.hs]: `pattern`

**Comment by virgil-serbanuta** [comment]
Thanks, all variable->pattern comments fixed.

**Pull request merged**